### PR TITLE
Fix embedding jvp support by making embedding_renorm ignore forward mode AD

### PR DESCRIPTION
**Pull request by samdow**
On functorch, we started seeing [embedding forward mode fail](https://github.com/pytorch/functorch/pull/816). From looking at it, we figured out that recently [embedding got forward mode support enabled](https://github.com/pytorch/pytorch/commit/369d9f4137a8bfc20e6a4e1d6ab35eeae4e9b345) and then doing forward mode with embedding and [max_norm doesn't work with gradcheck](https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_methods_invocations.py#L8877-L8881), so it's not checked.

What was happening is that `embedding_renorm` was setting `torch.no_grad()` which only turns off the backwards mode AD so functorch's jvp tests were still using forward mode AD during the `embedding_renorm` call. This makes it so that we don't use forward mode during the embedding_renorm call

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2050,8 +2050,13 @@ def hardswish(input: Tensor, inplace: bool = False) -> Tensor:
...
 
 def _no_grad_embedding_renorm_(weight: Tensor, input: Tensor, max_norm: float, norm_type: float) -> Tensor:
+    prev_state = get_fwd_grad_enabled()
     with torch.no_grad():
-        torch.embedding_renorm_(weight, input, max_norm, norm_type)
+        set_fwd_grad_enabled(False)
+        try:
+          torch.embedding_renorm_(weight, input, max_norm, norm_type)
+        finally:
+          set_fwd_grad_enabled(prev_state)
```
Comment (zou3519) [inline torch/nn/functional.py:2059]: Add test?

File: torch/nn/functional.py [hunk 2]
```diff
@@ -2050,8 +2050,13 @@ def hardswish(input: Tensor, inplace: bool = False) -> Tensor:
 
 
 def _no_grad_embedding_renorm_(weight: Tensor, input: Tensor, max_norm: float, norm_type: float) -> Tensor:
+    prev_state = get_fwd_grad_enabled()
     with torch.no_grad():
-        torch.embedding_renorm_(weight, input, max_norm, norm_type)
+        set_fwd_grad_enabled(False)
```
[the remark this answers is not in the document]
Comment (soulitzer) [reply torch/nn/functional.py:2055]: cc @albanD

[the remark this answers is not in the document]
Comment (samdow) [reply torch/nn/functional.py:2055]: Sorry for the confusion, Alban and I talked offline and this is a functorch-only function that I'm guessing worked locally through some dynamic linking or something?

This will change in a moment to use detach instead, though it might still be nice to have an equivalent for forward mode

**Comment by samdow** [comment]
Sorry again about the confusion, PR should be in final state of using detach and having tests. Locally, it fixes functorch and passes these tests (fwd ad one did fail before change to torch)

**Review by soulitzer — APPROVED** [review]
Thanks!

**Review by albanD — APPROVED** [review]
Thanks for the update!
